### PR TITLE
Ensured that Bar Chart dialog opens with the correct settings, and improved formatting

### PR DESCRIPTION
**Pull request by derekagorhom**
Fixes partly #8288 
This PR is ready for review
@rdstern i was able to fix items d, e, and g.
Item f will be fixed along with the new ones @COLIEWO found recently

### CODE REVIEW

**Comment by derekagorhom** [comment]
@COLIEWO ,  @rdstern the new changes solved the issues of item f and also the latest correction by celestine on` vjust` being changed from "5" to "1.5"

**Review by rdstern — CHANGES_REQUESTED** [review]
@derekagorhom this is looking good.  However, while you are there I looked at the translation into French.  Adding better width of labels would be good at the same time.
And @MeSophie could you please edit the database to improve the translation of a Pie chart - which is Tart and should be Camembert and also Donut seems to be Anneau.  @derekagorhom  please make sure the labels have enough space for these.

image

1. @derekagorhom you see the Reorgansier la - that (In English) is `Reorder Frequencies`  I suggest you get rif of the Frequencies here (Like the simple Reorder lower down.  Then the French will automatically be complete/
2. Towards the bottom make the label wider for the Save control label - there is plenty of space.
3. For the Pie (Tart) just make sure that the label is wide enough for when Sophie has added the correct Camembert translation.

**Comment by derekagorhom** [comment]
@rdstern I have made layout changes you requested, but Item a) will @lloyddewit to make database change in the `Id_text` for it to be complete.
I have also increased the length of the `pie` and `donut` checkboxes

**Review by rdstern — CHANGES_REQUESTED** [review]
@derekagorhom Here is a small problem, with the add.factor addition, so it can handle numeric data better.

```
# Code generated by the dialog, Bar Chart
diamonds <- data_book$get_data_frame(data_name="diamonds")
last_graph <- ggplot2::ggplot(diamonds, mapping=ggplot2::aes(x=forcats::fct_infreq(f=as.factor("carat")), fill="")) + ggplot2::geom_bar(stat="count") + theme_grey()
data_book$add_object(data_name="diamonds", object_name="last_graph", object_type_label="graph", object_format="image", object=check_graph(graph_object=last_graph))
data_book$get_object_data(data_name="diamonds", object_name="last_graph", as_file=TRUE)
```
In the code where you added as.factor, the variable is in quotes, here "carat".  Please delete the quotes.

**Review by N-thony — CHANGES_REQUESTED** [review]

**Comment by N-thony** [comment]
> @derekagorhom this is looking good. However, while you are there I looked at the translation into French. Adding better width of labels would be good at the same time. And @MeSophie could you please edit the database to improve the translation of a Pie chart - which is Tart and should be Camembert and also Donut seems to be Anneau. @derekagorhom please make sure the labels have enough space for these.
> 
> image
> 
> 1. @derekagorhom you see the Reorgansier la - that (In English) is `Reorder Frequencies`  I suggest you get rif of the Frequencies here (Like the simple Reorder lower down.  Then the French will automatically be complete/
> 2. Towards the bottom make the label wider for the Save control label - there is plenty of space.
> 3. For the Pie (Tart) just make sure that the label is wide enough for when Sophie has added the correct Camembert translation.

@derekagorhom can you make the checkbox `Réorganiser la` wider so that it can fit the french text.

**Comment by derekagorhom** [comment]
@N-thony , @rdstern I have fixed problem with the quotes occurring in the script.
Thanks

**Comment by N-thony** [comment]
> @N-thony , @rdstern I have fixed problem with the quotes occurring in the script and also increased the width of the checkbox for reorder frequency Thanks
@derekagorhom looks good now.

**Comment by N-thony** [comment]
> @N-thony I corrected the translation of `Pie` and `Donut` in Crowdin. And also corrected the translation of `Experiments` , I also wanted to change the translation of `File` to `Fichier` and not `déposer` (R-Instat Menu Items) but i received this notification. please can you have a look and approve the new translation? Thank you.
> 
> image

@MeSophie everything is approved already, I think @lloyddewit need to merge the new PRs created to update the database with new translations.

**Review by rdstern — DISMISSED** [review]
@derekagorhom well done - these now all seem to work.  It would be good to merge.  There will continue to be improvements, I am sure!  Over to you @lloyddewit

**Comment by lloyddewit** [comment]
@MeSophie Thank you for the CrowdIn changes. I accepted/approved them all. I also changed `File` to `Fichier`. You should see your updated translations later today when I merge the updated Json files into the database.

File: instat/sdgPlots.vb [hunk 1]
```diff
@@ -268,10 +268,10 @@ Public Class sdgPlots
 
         'Theme Tab Checkboxes under grpCommonOptions
         ucrChkLegendPosition.SetText("Legend Position")
-        ucrChkLegendPosition.AddToLinkedControls(ucrInputLegendPosition, {True}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True, bNewLinkedChangeToDefaultState:=True, objNewDefaultState:="None")
+        ucrChkLegendPosition.AddToLinkedControls(ucrInputLegendPosition, {True}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True, bNewLinkedChangeToDefaultState:=True, objNewDefaultState:="Left")
         ucrInputLegendPosition.SetDropDownStyleAsNonEditable()
         ucrInputLegendPosition.SetParameter(New RParameter("legend.position"))
-        dctLegendPosition.Add("None", Chr(34) & "none" & Chr(34))
+        'dctLegendPosition.Add("None", Chr(34) & "none" & Chr(34))
```
Comment (lloyddewit) [inline instat/sdgPlots.vb:274]:
```suggestion
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Review by lloyddewit — APPROVED** [review]

**Pull request merged**